tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java

```diff
@@ -61,11 +61,93 @@ public void testFunctionMocking() throws BallerinaTestException {
     public void testObjectMocking() throws BallerinaTestException {
         String msg1 = "7 passing";
         String msg2 = "6 failing";
+
+        String msg3 = "incorrect type of argument provided at position '1' to mock the function get()";
+
         String[] args = mergeCoverageArgs(new String[]{"object-mocking-tests"});
         String output = balClient.runMainAndReadStdOut("test", args,
                 new HashMap<>(), projectPath, false);
-        if (!output.contains(msg1) || !output.contains(msg2)) {
+        if (!output.contains(msg1) || !output.contains(msg2) || !output.contains(msg3)) {
             Assert.fail("Test failed due to object mocking failure in test framework.\nOutput:\n" + output);
         }
     }
+
+    @Test()
+    public void testObjectMocking_DefaultIncompatibleArgs() throws BallerinaTestException {
+        String msg = "incorrect type of argument provided at position '1' to mock the function get()";
+        String[] args = mergeCoverageArgs(new String[]{"--tests", "object_mocking:testDefaultIncompatibleArgs"});
+        String output =
+                balClient.runMainAndReadStdOut("test", args, new HashMap<>(),
+                        projectBasedTestsPath.resolve("object-mocking-tests").toString(), false);
+        if (!output.contains(msg)) {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
```

Comment:
Let's improve the error message to state what is failing. ATM, I'm noticing the same error message in multiple tests.

Reply:
The test cases are all condensed to one using data providers. I checked what the output for a test failure looks like and it shows which Ballerina test has failed so its not too difficult to identify. The failed test is specified in the output